src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketMultiBranchLinkActionFactory.java

```diff
@@ -44,7 +44,14 @@
     }
 
     @Override
-    public Class<Project> type() {
-        return Project.class;
+    public Class<WorkflowMultiBranchProject> type() {
+        return WorkflowMultiBranchProject.class;
+    }
+
+    private Optional<BitbucketSCMSource> getBitbucketSCMSource(WorkflowMultiBranchProject project) {
+        return project.getSCMSources().stream()
+                .filter(source -> source instanceof BitbucketSCMSource)
+                .map(source -> (BitbucketSCMSource) source)
+                .findAny();
```

Comment:
is it possible for users to hook up multiple BbS instances to their Jenkins instance? I am not totally familiar with BitbucketSCMSource but as it has `mirrorName` are there any implications of this

Reply:
This is an interesting question. Re. mirrors, no they will always be associated with the same instance. _However_ customers can have multiple branch sources. We've never really tested this, I assume you would do this if for example you had a forked workflow across instances that need to share a build plan.

In this case, the link doesn't make a lot of sense, and it might be better if we don't provide one, because which "bitbucket" to go to is completely ambiguous. That's a strong argument for omitting the link from the multibranch job entirely and just having it available on pipeline branches instead.

Reply:
Because we can have multiple branch sources it might be best to not provide a link if we can't break down that ambiguity. It would be interesting to see how the Branch Source Plugin (referenced in the ticket https://issues.jenkins-ci.org/browse/JENKINS-63070) navigates this and if they've come across similar issues.